_includes/components/breadcrumbs.html

```diff
@@ -8,8 +8,88 @@
 
 {%- if page.url != "/" and page.parent -%}
 
-  {%- assign pages_list = site[page.collection] | default: site.html_pages -%}
+{%- capture nav_list_link -%}
+<a href="{{ page.url | relative_url }}" class="nav-list-link">
+{%- endcapture -%}
+
+{%- capture site_nav -%}
+{%- include_cached components/site_nav.html -%}
+{%- endcapture -%}
+
+{%- if site_nav contains nav_list_link -%}
+
+  {%- capture nav_list_simple -%}
+  <ul class="nav-list">
+  {%- endcapture -%}
+
+  {%- capture nav_list_link_class %} class="nav-list-link">
+  {%- endcapture -%}
+
+  {%- capture nav_category -%}
+  <div class="nav-category">
+  {%- endcapture -%}
+
+  {%- assign nav_anchor_splits =
+        site_nav | split: nav_list_link |
+        first | split: nav_category |
+        last | split: "</a>" -%}
+
+  {%- comment -%}
+    The ordinary pages (if any) and the collections pages (if any) are separated by
+    occurrences of nav_category.
+    
+    Any ancestor nav-links of the page are contained in the last group of pages,
+    immediately preceding nav-lists. After splitting at "</a>", the anchor that
+    was split is a potential ancestor link when the following split starts with
+    a nav-list. 
+    
+    The array nav_breadcrumbs is the stack of current potential ancestors of the
+    current page. A split that contains one or more "</ul>"s requires that number
+    of potential ancestors to be popped from the stack.
+  {%- endcomment %}
+
+  {%- assign nav_breadcrumbs = "" | split: "" -%}
+
+  {%- for nav_split in nav_anchor_splits -%}
+  {%- unless forloop.last -%}
+
+  {%- assign nav_split_next = nav_anchor_splits[forloop.index] | trim -%}
+
+  {%- assign nav_split_test =
+        nav_split_next | remove_first: nav_list_simple | prepend: nav_list_simple -%}
+  {%- if nav_split_test == nav_split_next -%}
+    {%- assign nav_breadcrumb_link =
+          nav_split | split: "<a " | last | prepend: "<a " |
+          replace: nav_list_link_class, ">" | append: "</a>" -%}
+    {%- assign nav_breadcrumbs = nav_breadcrumbs | push: nav_breadcrumb_link -%}
+  {%- endif -%}
+
+  {%- if nav_split_next contains "</ul>" -%}
+    {%- assign nav_endless = nav_split_next | remove: "</ul>" -%}
+    {%- assign nav_ends =
+          nav_split_next.size | minus: nav_endless.size | divided_by: 5 -%}
```

Comment:
Quick question: why is the `divided_by: 5` filter here? I'm not sure that I understand what `nav_ends` is supposed to be.

Reply:
Indeed, that should have been explained. Is it adequate to add the following in the preceding comment?

```
    To compute the number of "</ul>" strings in nav_split_next, removing them all 
    gives the string nav_endless. The difference between the sizes of 
    nav_split_next and nav_endless is the number of removed strings times 5.
```

A corresponding comment in `_includes/css/activation.scss.liquid` could be:

```
  To compute the number of nav_list strings in a nav_split, removing them all 
  gives the string nav_listless. The difference between the sizes of nav_split 
  and nav_listless is the number of removed strings times their size.
  Similarly to compute nav_ends, where the size of the removed strings is 5.
```